yt_dlp/extractor/instagram.py

```diff
@@ -403,17 +420,6 @@ def _real_extract(self, url):
             self.report_warning('General metadata extraction failed (some metadata might be missing).', video_id)
         media.update(traverse_obj(general_info, ('data', 'shortcode_media')) or {})
 
-        if self._get_cookies(url).get('sessionid'):
```

Comment:
why remove this? this is how we determine is the user is logged-in or not

Reply:
What if sessionid used expires? So the indicator used here is the info api, if it gets resolved then the user is logged in and if it doesn't then the user is not logged in. But the workflow is as mentioned by you, with an exception `/embed/captioned` not making into the code.

Reply:
if the user is logged in we want them to bypass the `get_ruling_for_content` API step, and the `sessionid` cookie is the best indicator we have of whether the user is authenticated. if the cookie is expired, then the `media` API call will fail and the extractor can fall back on the logged-out flow (graphql -> webpage)

Reply:
Isn't the current code doing the same thing without the `sessionid` check?

Reply:
as i just said above: if the user is logged in we want them to bypass the `get_ruling_for_content` API step

we also don't want logged-out users sending API calls that will fail and work against their rate-limit

Reply:
you are correct. looks ok to me now

Reply:
But for cosmetic purposes i can add it anyway. The diff. is that just the failed media API call won't be made for non logged in users.

Reply:
> Isn't the current code doing the same thing without the `sessionid` check?

logged-out users are sending an unnecessary and counter-productive request to the mobile API (`i.instagram.com//media/{media_pk}/info/`) before the api_check graphql request. this is working against their rate-limit. the `sessionid` check should happen before the mobile API request (which is only useful for logged-in users)